src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/18.diff.sql

```diff
@@ -0,0 +1,374 @@
+            AND dbo.ExtendedQueryTag.TagStatus <> 2
+        ) AS S
+        ON T.ResourceType = @resourceType
+            AND T.TagKey = S.TagKey
+            AND T.PartitionKey = @partitionKey
+            AND T.SopInstanceKey1 = @studyKey
+            AND ISNULL(T.SopInstanceKey2, @seriesKey) = @seriesKey
+            AND ISNULL(T.SopInstanceKey3, @instanceKey) = @instanceKey
+        WHEN MATCHED AND @watermark > T.Watermark THEN
+            -- When index already exist, update only when watermark is newer
+            UPDATE SET T.Watermark = @watermark, T.TagValue = S.TagValue
+        WHEN NOT MATCHED THEN
+            INSERT (TagKey, TagValue, PartitionKey, SopInstanceKey1, SopInstanceKey2, SopInstanceKey3, Watermark, ResourceType)
+            VALUES
+            (
+                S.TagKey,
+                S.TagValue,
+                @partitionKey,
+                @studyKey,
+                (CASE WHEN S.TagLevel <> 2 THEN @seriesKey ELSE NULL END),
+                (CASE WHEN S.TagLevel = 0 THEN @instanceKey ELSE NULL END),
+                @watermark,
+                @resourceType
+            );
+    END
+
+    -- DateTime Key tags
+    IF EXISTS (SELECT 1 FROM @dateTimeExtendedQueryTags)
+    BEGIN
+        MERGE INTO dbo.ExtendedQueryTagDateTime AS T
+        USING
+        (
+            SELECT input.TagKey, input.TagValue, input.TagValueUtc, input.TagLevel
+            FROM @dateTimeExtendedQueryTags input
+           INNER JOIN dbo.ExtendedQueryTag WITH (REPEATABLEREAD)
+            ON dbo.ExtendedQueryTag.TagKey = input.TagKey
+            -- Only merge on extended query tag which is being added
+            AND dbo.ExtendedQueryTag.TagStatus <> 2
+        ) AS S
+        ON T.ResourceType = @resourceType
+            AND T.TagKey = S.TagKey
+            AND T.PartitionKey = @partitionKey
+            AND T.SopInstanceKey1 = @studyKey
+            AND ISNULL(T.SopInstanceKey2, @seriesKey) = @seriesKey
+            AND ISNULL(T.SopInstanceKey3, @instanceKey) = @instanceKey
+        WHEN MATCHED AND @watermark > T.Watermark THEN
+            -- When index already exist, update only when watermark is newer
+            UPDATE SET T.Watermark = @watermark, T.TagValue = S.TagValue
+        WHEN NOT MATCHED THEN
+            INSERT (TagKey, TagValue, PartitionKey, SopInstanceKey1, SopInstanceKey2, SopInstanceKey3, Watermark, TagValueUtc, ResourceType)
+            VALUES
+            (
+                S.TagKey,
+                S.TagValue,
+                @partitionKey,
+                @studyKey,
+                (CASE WHEN S.TagLevel <> 2 THEN @seriesKey ELSE NULL END),
+                (CASE WHEN S.TagLevel = 0 THEN @instanceKey ELSE NULL END),
+                @watermark,
+                S.TagValueUtc,
+                @resourceType
+            );
+    END
+
+    -- PersonName Key tags
+    IF EXISTS (SELECT 1 FROM @personNameExtendedQueryTags)
+    BEGIN
+        MERGE INTO dbo.ExtendedQueryTagPersonNam AS T
```

Comment:
Oops! Typo and I should actually use this new version by bumping the max version